app/code/Magento/Inventory/Ui/Component/Control/Stock/GenericButton.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Ui\Component\Control\Stock;
+
+use Magento\Backend\Block\Widget\Context;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+/**
+ * Class GenericButton
+ */
+class GenericButton
+{
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
```

Comment:
dependency on Context is not recommended. As Context is a container for other objects.

Use dependencies on the precise components you are dependent on